desc/magnetic_fields/_core.py

```diff
@@ -2644,60 +2666,48 @@ def field_line_integrate(
     min_step_size = jnp.where(
         phis[-1] > phis[0], min_step_size, -jnp.abs(min_step_size)
     )
-
-    @jit
-    def odefun(s, rpz, args):
-        rpz = rpz.reshape((3, -1)).T
-        field = args[0]
-        r = rpz[:, 0]
-        br, bp, bz = (
-            scale
-            * field.compute_magnetic_field(
-                rpz, params, basis="rpz", source_grid=source_grid, chunk_size=chunk_size
-            ).T
-        )
-        return jnp.array(
-            [r * br / bp * jnp.sign(bp), jnp.sign(bp), r * bz / bp * jnp.sign(bp)]
-        ).squeeze()
+    max_steps = (
+        max_steps
+        if max_steps is not None
+        else int((phis[-1] - phis[0]) / min_step_size * 1000)
```

Comment:
this would make the default max_steps ~ 1e11 which is likely way too huge. Also note that the `RecursiveCheckpointAdjoint` means that the integration loop is actually done as a for loop + conditional in the body, so it will always do the maximum number of steps (but with a no-op for steps after convergence) but that still takes time, especially for ~1e11 steps. 

It seems like in most cases it takes ~<100 steps per transit, so I think we could still be pretty safe if we just did max_steps=~500-1000 steps per transit

Reply:
use less steps here, let particle tracing have different default